front/components/assistant/conversation/input_bar/editor/MentionWithPaste.tsx

```diff
@@ -18,7 +18,11 @@ export const MentionWithPaste = Mention.extend({
             return [
               ...text.matchAll(
                 // Note: matching the @ that are found either at the start of a line or after a whitespace character.
-                new RegExp("(^@|\\s@)" + escapeRegExp(suggestion.label), "g")
+                // and that also are followed by a newline, a whitespace character or the end of the string.
+                new RegExp(
+                  "(^@|\\s@)" + escapeRegExp(suggestion.label) + "(\\n|\\s|$)",
```

Comment:
I guess `\s` should be enough here, right?
```suggestion
                  "(^@|\\s@)" + escapeRegExp(suggestion.label) + "(\\s|$)",
```

Reply:
Also not need to capture the group (you can prefix with `?:`.